services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_mode_dynamic.py

```diff
@@ -0,0 +1,119 @@
+from fastapi import FastAPI
+from models_library.docker import DockerLabelKey
+from models_library.generated_models.docker_rest_api import Node, Task
+from servicelib.logging_utils import LogLevelInt
+
+from ..core.settings import get_application_settings
+from ..models import AssociatedInstance, EC2InstanceData, EC2InstanceType, Resources
+from ..utils import dynamic_scaling as utils
+from ..utils import ec2, utils_docker
+from ..utils.rabbitmq import log_tasks_message, progress_tasks_message
+from .auto_scaling_mode_base import BaseAutoscaling
+from .docker import get_docker_client
+
+
+class DynamicAutoscaling(BaseAutoscaling):
```

Comment:
So you have dynamic and computational implemntations?

Reply:
yes, but this comes in the next PR